types/types.go

```diff
@@ -475,9 +476,30 @@ func NewFunction(args []Type, result Type) *Function {
 	}
 }
 
-// Args returns the function's argument types.
+// NewVariadicFunction returns a new Function object. This function sets the
+// variadic bit on the signature. Non-void variadic functions are not currently
+// supported.
+func NewVariadicFunction(args []Type, varargs Type, result Type) *Function {
+	if result != nil {
+		panic("illegal value: non-void variadic functions not supported")
+	}
+	return &Function{
+		args:     args,
+		variadic: varargs,
+		result:   nil,
+	}
+}
+
+// FuncArgs returns the function's arguments.
+func (t *Function) FuncArgs() FuncArgs {
+	return FuncArgs{Args: t.Args(), Variadic: t.variadic}
+}
+
+// Args is deprecated. Use FuncArgs instead.
```

Comment:
I'm not 100% certain, but I think the format that'll have tooling pick up the deprecation would be:
```suggestion
// Args returns the functions args, ignoring variadic arguments.
// Deprecated: Use FuncArgs instead.
```